bigquery_etl/pytest_plugin/sql.py

```diff
@@ -123,6 +124,20 @@ def runtest(self):
                         table_name.replace(".", "_").replace("-", "_"),
                     )
                     query = query.replace(original, table_name)
+
+                # second check for tablename tweaks.
+                # if the tablename ends with a date then need to replace that date with '*' for the
+                # query text substitution to work.
+                # e.g. see moz-fx-data-marketing-prod.65789850.ga_sessions_20230214
+                # A query using that table uses moz-fx-data-marketing-prod.65789850.ga_sessions_*
+                # with the date appended to allow for daily processing.
+                try:
+                    datetime.datetime.strptime(table_name[-8:], "%Y%m%d")
+                    generic_table_name = table_name[:-8] + "*"
+                    generic_original = original[:-8] + "*"
+                    query = query.replace(generic_original, generic_table_name)
+                except ValueError:
+                    pass
```

Comment:
lgtm, but should probably take advantage of try/catch/else to only catch exceptions from the intended line:
```suggestion
                try:
                    datetime.datetime.strptime(table_name[-8:], "%Y%m%d")
                except ValueError:
                    pass
                else:
                    generic_table_name = table_name[:-8] + "*"
                    generic_original = original[:-8] + "*"
                    query = query.replace(generic_original, generic_table_name)
```